### DISCUSSION TITLE
Rename OS X to macOS in source code

### DISCUSSION THREAD

**Discussion by hbugdoll** [appcelerator/titanium_mobile#13496] [General]
In my opinion all `osx` occurrences in source code should be renamed to `macos`.
The last OS X major version was realeased 7 years ago.

Example: realease asset file for macOS is still named `mobilesdk-*.*.*.**-osx.zip`

**Comment by cb1kenobi** [comment]
It's a good idea... a bit of work given we need backwards compatibility. Let's do it!

**Comment by caspahouzer** [comment]
@hbugdoll wanna contribute with some changes? Take a look at https://tidev.io/contribute

**Comment by hbugdoll** [comment]
I now see three key points:

1. ZIP filename `mobilesdk-*.*.*.**-osx.zip`
    - Changing it in GitHub Actions & Workflows is easy, but need a change in CLI too.
    - Should the CLI rename the local directory `/Users/$(USER)/Library/Application Support/Titanium/mobilesdk/osx/` to `/Users/$(USER)/Library/Application Support/Titanium/mobilesdk/macos/` or should both directories exist in parallel with only new SDKs in the new directory? In the latter case, however, backward compatibility is required.
2. `TITANIUM_SDK = /Users/$(USER)/Library/Application Support/Titanium/mobilesdk/osx/$(TITANIUM_SDK_VERSION)` in `.xcconfig` files
    - Changing this ad hoc in all modules would be too cumbersome, backwards compatibility should be created in the CLI here (see 1.).
3. config keys `osx.executables.security` and `osx.executables.xcodeSelect`
    - CLI could rename both keys
    - used also by old `node-appc` (is migrating to `node-titanium-sdk`) and `appc-daemon` (outdated?)

**Reply by cb1kenobi** [reply]
Regarding the CI, when a new build is created, it registers it with the downloads site like this: https://downloads.titaniumsdk.com/registry/master.json. We should be able to change the filename with little issue, but changing the `"os"` will break all older CLIs. We could go back and rewrite the registry files adding a 4th platform just for `"macos"`, then in a year we could drop the "osx" platform. Seems like a lot of unnecessary work for an internal thing.

Renaming the install path should be doable. The path will be owned by `node-titanium-sdk` eventually and the rewrite is not ready.

Honestly, we may need to put a pin in this for a couple months, but we can plan for it in `node-titanium-sdk` and ship a new CLI first.
